server/application/application.proto

```diff
@@ -35,6 +35,29 @@ message ApplicationQuery {
 	optional string appNamespace = 7;
 	// the project names to restrict returned list applications (legacy name for backwards-compatibility)
 	repeated string project = 8;
+	// the application name to start from (app with min name is included in response)
+	optional string minName = 9;
+	// the application name to end at (app with max name is included in response)
+	optional string maxName = 10;
+	// the repos filter, ignore if the field repo was defined
+	repeated string repos = 11;
+	// the clusters filter
+	repeated string clusters = 12;
+	// the namespaces filter
+	repeated string namespaces = 13;
+	// the auth sync filter
+	optional bool autoSyncEnabled = 14;
+	// the sync status filter
+	repeated string syncStatuses = 15;
+	// the health status filter
+	repeated string healthStatuses = 16;
+	// search
+	optional string search = 17;
+  	// offset
+	optional int64 offset = 18;
+	// limit
+	optional int64 limit = 19;
+
```

Comment:
Although it's not listed in the proposal, maybe we should implement server-side sorting when adding server-side pagination as well.

![image](https://github.com/user-attachments/assets/289de86b-85e7-46b4-9247-a11e47d361b1)

Reply:
omg